gazebo_plugins/src/gazebo_ros_camera.cpp

```diff
@@ -447,14 +447,14 @@ void GazeboRosCamera::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _
 
     // camera_ projection matrix (same as camera_ matrix due
     // to lack of distortion/rectification) (is this generated?)
-    camera_info_msg.p[0] = focal_length;
+    camera_info_msg.p[0] = _sdf->Get<double>("fx_p", focal_length).first;
     camera_info_msg.p[1] = 0.0;
-    camera_info_msg.p[2] = cx;
-    camera_info_msg.p[3] = -focal_length * hack_baseline;
+    camera_info_msg.p[2] = _sdf->Get<double>("cx'", cx).first;
```

Comment:
Is a single-quote character valid in SDF? I'm trying to imagine what setting this in practice looks like,

```xml
<cx'>100</cx'>
```

I would make the names clear that they're specifically for overriding the projection matrix, P, e.g. `P_cx`, `P_cy`, `P_fx`, `P_fy`, etc.

Reply:
Yup that'd be a good idea. I've pushed changes to reflect that.